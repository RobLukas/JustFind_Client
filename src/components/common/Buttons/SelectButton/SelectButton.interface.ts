interface SelectButtonProps {
  collection: string[];
  value: string;
  onChange(e: string): void;
}

export default SelectButtonProps;
