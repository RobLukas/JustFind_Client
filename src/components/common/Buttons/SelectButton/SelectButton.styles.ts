import styled from 'styled-components';

const Select = styled.select`
  display: inherit;
  color: ${({ theme }) => theme.styles.primary.color};
  background: ${({ theme }) => theme.styles.secondary.background};
  font-size: 18px;
  border-radius: 3px;
  padding: 15px;
  border: 0;
  outline: 0;
  -webkit-appearance: none;
  width: 100%;
  height: 100%;
`;

export default {
  Select,
};
