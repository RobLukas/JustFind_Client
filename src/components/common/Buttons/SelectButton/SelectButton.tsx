import React, { FC } from 'react';
import { v4 as uuid } from 'uuid';

import SelectButtonProps from './SelectButton.interface';
import S from './SelectButton.styles';

const SelectButton: FC<SelectButtonProps> = ({
  collection,
  value,
  onChange,
}: SelectButtonProps) => {
  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) =>
    onChange(e.target.value);

  return (
    <S.Select onChange={handleChange} value={value}>
      {collection.map((item) => (
        <option key={uuid()}>{item}</option>
      ))}
    </S.Select>
  );
};

export default SelectButton;
