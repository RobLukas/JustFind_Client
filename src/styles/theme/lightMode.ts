export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
