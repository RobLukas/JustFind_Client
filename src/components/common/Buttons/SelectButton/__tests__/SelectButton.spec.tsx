import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import lightMode from 'styles/theme/lightMode';
import SelectButton from '../SelectButton';

describe('<SelectButton />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const collection = ['1', '2'];
    const onChange = jest.fn();
    const value = collection[0];

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <SelectButton
          value={value}
          collection={collection}
          onChange={onChange}
        />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
