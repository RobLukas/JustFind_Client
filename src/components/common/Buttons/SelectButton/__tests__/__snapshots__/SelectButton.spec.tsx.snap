// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectButton /> render correctly 1`] = `
.c0 {
  display: inherit;
  color: #37474f;
  background: #fff;
  cursor: pointer;
  font-size: 18px;
  border-radius: 3px;
  padding: 15px;
  border: 0;
  outline: 0;
  -webkit-appearance: none;
  width: 100%;
  height: 100%;
}

<select
  class="c0"
>
  <option>
    1
  </option>
  <option>
    2
  </option>
</select>
`;
